repository.go

```diff
@@ -165,6 +165,18 @@ func (v *Repository) Config() (*Config, error) {
 	return config, nil
 }
 
+func (v *Repository) SetConfig(c *Config) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_repository_set_config(v.ptr, c.ptr)
+	runtime.KeepAlive(v)
```

Comment:
```suggestion
	runtime.KeepAlive(v)
	runtime.KeepAlive(c)
```

otherwise if `c` is not used by the caller later, the Go runtime is free to remove it in the middle of the CGO call.